src/numeric/splines/csplines.jl

```diff
@@ -21,78 +23,34 @@ d2Ad = [
    [ 0.0 0.0  1.0  0.0]
 ]
 
-function eval_UC_spline(smin, smax, orders, C, S)
-
-    d = size(S,2)
-    N = size(S,1)
 
-    vals = zeros(N)
-    eval_UC_spline!(smin, smax, orders, C, S, vals)
-    return vals
+# old style call
+function eval_UC_spline(a, b, orders, C::Array{Float64}, S::Matrix{Float64})
+    d,N = size(S)
 
+    n_x = size(C,1)
+    dims = tuple(size(C)[2:end]...)
+    out = zeros(n_x,N)
+    CC = reinterpret(SVector{n_x,Float64},C,dims)
+    SS = reinterpret(SVector{d,Float64},S,(N,))
+    V  = reinterpret(SVector{n_x,Float64},out,(N,))
+    eval_UC_spline!(a, b, orders, CC, SS, V)
+    return out
 end
 
-function eval_UC_spline_G(a, b, orders, C, S)
-
-    d = size(S,2)
-    N = size(S,1)
-
-    vals = zeros(N)
-    grad = zeros(N,d)
-
-    eval_UC_spline_G!(a, b, orders, C, S, vals, grad)
 
-    return (vals, grad)
-
-end
-
-function eval_UC_multi_spline(a, b, orders, C, S)
-
-    d = size(S,2)
-    N = size(S,1)
-    K = size(C,1) # number of splines to evaluate
-    vals = zeros(K,N)
-    eval_UC_multi_spline!(a, b, orders, C, S, vals)
+function eval_UC_spline(a, b, orders, C::Array{T, d}, S::Vector{SVector{d,Float64}}) where T where d
+    N = length(S)
+    vals = zeros(T,N)
+    eval_UC_spline!(a, b, orders, C, S, vals)
```

Comment:
it is convention to put the mutated argument as the first argument.

We should probably follow the convention.

I don't really like that myself, but it's the way things are...

Reply:
Sure. If you don't mind, I'll fix that in the upcoming PR.